// @cypress/json-schemas/dist/schemas/commit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var schema_tools_1 = require("@cypress/schema-tools");
exports.commitExample100 = {
    authorEmail: 'me@company.com',
    authorName: 'Joe Doe',
    branch: 'master',
    message: 'greatest feature ever',
    sha: '69cc7efa6b8fec8896debd4b3a2fae2a6120d431',
    url: 'https://github.com/username/repo/commit/69cc7ef',
};
exports.Commit100 = {
    version: {
        major: 1,
        minor: 0,
        patch: 0,
    },
    schema: {
        title: 'commit',
        type: 'object',
        description: 'Commit information returned by the API',
        properties: {
            authorEmail: {
                type: ['string', 'null'],
                description: 'Usually author email',
            },
            authorName: {
                type: ['string', 'null'],
                description: 'Author name, like Joe Smith',
            },
            branch: {
                type: ['string', 'null'],
                description: 'Git branch',
            },
            message: {
                type: ['string', 'null'],
                description: 'Commit message',
            },
            sha: {
                type: ['string', 'null'],
                description: 'Typically a commit SHA',
            },
            url: {
                type: ['string', 'null'],
                description: 'Optional source commit url',
            },
        },
        required: true,
        additionalProperties: false,
    },
    example: exports.commitExample100,
};
exports.Commit = schema_tools_1.versionSchemas(exports.Commit100);
