// @cypress/json-schemas/dist/schemas/test-runner-commit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var schema_tools_1 = require("@cypress/schema-tools");
exports.testRunnerCommitExample100 = {
    authorEmail: 'me@company.com',
    authorName: 'Joe Doe',
    branch: 'master',
    message: 'greatest feature ever',
    sha: '69cc7efa6b8fec8896debd4b3a2fae2a6120d431',
    remoteOrigin: null,
};
var TestRunnerCommit100 = {
    version: {
        major: 1,
        minor: 0,
        patch: 0,
    },
    schema: {
        title: 'testRunner.commit',
        type: 'object',
        description: 'Commit information sent by the test runner',
        properties: {
            authorEmail: {
                type: ['string', 'null'],
                description: 'Usually author email',
            },
            authorName: {
                type: ['string', 'null'],
                description: 'Author name, like Joe Smith',
            },
            branch: {
                type: ['string', 'null'],
                description: 'Git branch',
            },
            message: {
                type: ['string', 'null'],
                description: 'Commit message',
            },
            sha: {
                type: ['string', 'null'],
                description: 'Typically a commit SHA',
            },
            remoteOrigin: {
                type: ['string', 'null'],
                description: 'Remote Git origin url',
            },
        },
        required: true,
        additionalProperties: false,
    },
    example: exports.testRunnerCommitExample100,
};
exports.TestRunnerCommit110 = schema_tools_1.addProperty({
    schema: TestRunnerCommit100,
    description: 'Commit with pull request information',
}, {
    property: 'pullRequestId',
    propertyType: ['string', 'null'],
    propertyFormat: null,
    exampleValue: '797',
    propertyDescription: 'Pull request number',
    isRequired: false,
}, {
    property: 'defaultBranch',
    propertyType: ['string', 'null'],
    propertyFormat: null,
    exampleValue: 'develop',
    propertyDescription: 'Default branch for the build / pull request / pipeline',
    isRequired: false,
});
exports.TestRunnerCommit = schema_tools_1.versionSchemas(TestRunnerCommit100, exports.TestRunnerCommit110);
