// @cypress/json-schemas/dist/schemas/test-runner-cypress-stats.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var schema_tools_1 = require("@cypress/schema-tools");
exports.testRunnerCypressStatsExample100 = {
    suites: 5,
    tests: 6,
    passes: 1,
    pending: 1,
    failures: 3,
    skipped: 0,
    wallClockStartedAt: '2018-02-01T20:14:19.323Z',
    wallClockEndedAt: '2018-02-01T20:14:19.323Z',
    wallClockDuration: 1234,
};
exports.TestRunnerCypressStats100 = {
    version: {
        major: 1,
        minor: 0,
        patch: 0,
    },
    schema: {
        title: 'testRunner.cypressStats',
        type: 'object',
        description: 'Test stats Cypress test runner collects and sends',
        properties: {
            suites: {
                type: 'integer',
                minimum: 0,
            },
            tests: {
                type: 'integer',
                minimum: 0,
            },
            passes: {
                type: 'integer',
                minimum: 0,
            },
            pending: {
                type: 'integer',
                minimum: 0,
            },
            failures: {
                type: 'integer',
                minimum: 0,
            },
            skipped: {
                type: 'integer',
                minimum: 0,
                description: 'Number of tests that were skipped due to errors in "before" / "beforeEach" hooks',
            },
            wallClockStartedAt: {
                type: 'string',
                format: 'date-time',
            },
            wallClockEndedAt: {
                type: 'string',
                format: 'date-time',
            },
            wallClockDuration: {
                type: 'integer',
                minimum: 0,
                description: 'ms',
            },
        },
        required: true,
        additionalProperties: false,
    },
    example: exports.testRunnerCypressStatsExample100,
};
exports.TestRunnerCypressStats = schema_tools_1.versionSchemas(exports.TestRunnerCypressStats100);
