// @cypress/json-schemas/dist/schemas/test-runner-ci.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var schema_tools_1 = require("@cypress/schema-tools");
exports.testRunnerCiExample100 = {
    provider: 'CircelCI',
    buildNumber: '123',
    params: {
        accountName: 'foo',
    },
};
exports.TestRunnerCi100 = {
    version: {
        major: 1,
        minor: 0,
        patch: 0,
    },
    schema: {
        title: 'testRunner.ci',
        type: 'object',
        description: 'CI information sent by the test runner',
        properties: {
            provider: {
                type: ['string', 'null'],
                description: 'name of the CI provider, like "Circle"',
            },
            buildNumber: {
                type: ['string', 'null'],
                description: 'CI issues build number',
            },
            params: {
                type: ['object', 'null'],
                description: 'Any additional CI parameters',
            },
        },
        required: true,
        additionalProperties: false,
    },
    example: exports.testRunnerCiExample100,
};
exports.TestRunnerCi = schema_tools_1.versionSchemas(exports.TestRunnerCi100);
